core/src/main/java/hudson/cli/CLICommand.java

```diff
@@ -344,8 +347,16 @@ public Channel checkChannel() throws AbortException {
     @Deprecated
     protected Authentication loadStoredAuthentication() throws InterruptedException {
         try {
-            if (channel!=null)
-                return new ClientAuthenticationCache(channel).get();
+            if (channel!=null){
+                Authentication authLoadedFromCache = new ClientAuthenticationCache(channel).get();
+
+                if(!Functions.isAnonymous(authLoadedFromCache)){
+                    UserDetails userDetails = new CLIUserDetails(authLoadedFromCache);
+                    SecurityListener.fireAuthenticated(userDetails);
```

Comment:
Makes [this Javadoc](https://github.com/jenkinsci/jenkins/blob/bcbd5ba56ad7bbed00977b2d733f60019c6af04c/core/src/main/java/jenkins/security/SecurityListener.java#L50) inaccurate.